tests/rule_based_profiler/parameter_builder/test_parameter_builder.py

```diff
@@ -0,0 +1,486 @@
+)
+def test_parameter_builder_dependencies_evaluated_in_parameter_builder_if_not_precomputed(
+    mock_convert_to_json_serializable: mock.MagicMock,
+):
+    my_evaluation_dependency_0_parameter_builder: ParameterBuilder = (
+        DummyParameterBuilder(
+            name="my_evaluation_dependency_parameter_name_0",
+            evaluation_parameter_builder_configs=None,
+        )
+    )
+    my_evaluation_dependency_1_parameter_builder: ParameterBuilder = (
+        DummyParameterBuilder(
+            name="my_evaluation_dependency_parameter_name_1",
+            evaluation_parameter_builder_configs=None,
+        )
+    )
+
+    evaluation_parameter_builder_configs: Optional[List[ParameterBuilderConfig]] = [
+        ParameterBuilderConfig(
+            **my_evaluation_dependency_0_parameter_builder.to_json_dict()
+        ),
+        ParameterBuilderConfig(
+            **my_evaluation_dependency_1_parameter_builder.to_json_dict(),
+        ),
+    ]
+    my_dependent_parameter_builder: ParameterBuilder = DummyParameterBuilder(
+        name="my_dependent_parameter_builder",
+        evaluation_parameter_builder_configs=evaluation_parameter_builder_configs,
+    )
+
+    domain = DummyDomain()
+
+    parameter_container = ParameterContainer(parameter_nodes=None)
+    parameters: Dict[str, ParameterContainer] = {
+        domain.id: parameter_container,
+    }
+
+    all_fully_qualified_parameter_names: List[
+        str
+    ] = get_fully_qualified_parameter_names(
+        domain=domain,
+        variables=None,
+        parameters=parameters,
+    )
+
+    """
+    This assertion shows that neither "ParameterBuilder" dependencies computed their values, and dependent
+    "ParameterBuilder" also has not yet computed its value.
+    """
+    assert not all_fully_qualified_parameter_names
+
+    my_dependent_parameter_builder.resolve_evaluation_dependencies(
+        domain=domain,
+        variables=None,
+        parameters=parameters,
+        fully_qualified_parameter_names=all_fully_qualified_parameter_names,
+        runtime_configuration=None,
+    )
+
+    dependencies_fully_qualified_parameter_names: List[
+        str
+    ] = get_fully_qualified_parameter_names(
+        domain=domain,
+        variables=None,
+        parameters=parameters,
+    )
+
+    """
+    This assertion shows that because namespace of fully_qualified_parameter_names is empty (as previously asserted),
+    and because "ParameterBuilder" dependencies were specified as argument to instantiation of dependent
+    "ParameterBuilder", both "ParameterBuilder" dependencies computed their values (this was accomplished by
+    "my_dependent_parameter_builder.resolve_evaluation_dependencies()" which ran "_build_parameters()" of each
+    dependency "ParameterBuilder" object).  However, dependent "ParameterBuilder" has not yet computed its value (as
+    name "my_dependent_parameter_builder" is not in list), since it has yet been called.
+    """
```

Comment:
Could this more plainly be stated as "ensuring that test setup was successful?"

Where are the tests for resolve_evaluation_dependencies? These kinds of explanations, if required, belong there. I'd consider breaking this into its own test and potential local fixture rather than copy/pasting these `resolve_evaluation_dependencies` tests into each of the parameter builder tests

Reply:
@donaldheppner This is more than just set up.  Please see my earlier response in `https://github.com/great-expectations/great_expectations/pull/8302#discussion_r1263799594` about the place of `resolve_evaluation_dependencies()` -- still, your suggestion of testing it once, with its own fixture is very appealing.  Should we go ahead then and refactor as such (even if it requires its own shared memory setup)?  Thank you.